src/Time/TimeStamp.hs

```diff
@@ -49,9 +74,75 @@ timeMul :: forall (unit :: Rat) . KnownRat unit
         -> Time unit
 timeMul n (Time t) = Time (n * t)
 
+{- | Operator version of 'timeMul'.
+
+>>> 3 *:* sec 5
```

Comment:
I think you should add `infixl` or `infixr` for this operator with some priority so something like this would be possible to write:

```haskell
ghci> 2 *:* 3 *:* sec 5
ghci> 3 *:* 5 *:* 7 :: Time Second
```

Not sure it's possible but looks possible...